ui/components/app/modals/deposit-token-modal/deposit-token-modal.component.js

```diff
@@ -0,0 +1,184 @@
+      address: PropTypes.string.isRequired,
+      decimals: PropTypes.number,
+      symbol: PropTypes.string,
+      image: PropTypes.string,
+      aggregators: PropTypes.array,
+      isERC721: PropTypes.bool,
+    }),
+    chainId: PropTypes.string.isRequired,
+    isTestnet: PropTypes.bool.isRequired,
+    isBuyableCoinbasePayChain: PropTypes.bool.isRequired,
+    isBuyableTransakChain: PropTypes.bool.isRequired,
+    isTokenBuyableCoinbasePay: PropTypes.bool.isRequired,
+    isTokenBuyableTransak: PropTypes.bool.isRequired,
+    toCoinbasePay: PropTypes.func.isRequired,
+    toTransak: PropTypes.func.isRequired,
+    address: PropTypes.string.isRequired,
+    toFaucet: PropTypes.func.isRequired,
+    hideWarning: PropTypes.func.isRequired,
+    hideModal: PropTypes.func.isRequired,
+    showAccountDetailModal: PropTypes.func.isRequired,
+  };
+
+  goToAccountDetailsModal = () => {
+    this.props.hideWarning();
+    this.props.hideModal();
+    this.props.showAccountDetailModal();
+  };
+
+  renderRow({
+    logo,
+    title,
+    text,
+    buttonLabel,
+    onButtonClick,
+    hide,
+    className,
+    hideButton,
+    hideTitle,
+    onBackClick,
+    showBackButton,
+  }) {
+    if (hide) {
+      return null;
+    }
+
+    return (
+      <div className={className || 'deposit-ether-modal__buy-row'}>
+        {onBackClick && showBackButton && (
+          <div
+            className="deposit-ether-modal__buy-row__back"
+            onClick={onBackClick}
+          >
+            <i className="fa fa-arrow-left cursor-pointer" />
+          </div>
+        )}
+        <div className="deposit-ether-modal__buy-row__logo-container">
+          {logo}
+        </div>
+        <div className="deposit-ether-modal__buy-row__description">
+          {!hideTitle && (
+            <div className="deposit-ether-modal__buy-row__description__title">
+              {title}
+            </div>
+          )}
+          <div className="deposit-ether-modal__buy-row__description__text">
+            {text}
+          </div>
+        </div>
+        {!hideButton && (
+          <div className="deposit-ether-modal__buy-row__button">
+            <Button
+              type="secondary"
+              className="deposit-ether-modal__deposit-button"
+              large
+              onClick={onButtonClick}
+            >
+              {buttonLabel}
+            </Button>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  render() {
+    const {
+      chainId,
+      toCoinbasePay,
+      toTransak,
+      address,
+      toFaucet,
+      isTestnet,
+      isBuyableCoinbasePayChain,
+      isBuyableTransakChain,
+      isTokenBuyableCoinbasePay,
+      isTokenBuyableTransak,
+    } = this.props;
+    const { t } = this.context;
+    const networkName = NETWORK_TO_NAME_MAP[chainId];
+    const { symbol } = this.props.token;
+
+    return (
+      <div className="page-container page-container--full-width page-container--full-height">
```

Comment:
I 100% agree with your take, and i even considered myself. Reason why i didn't go down this path is that we were actually aiming for the "low hanging fruit" which was enabling token purchasing. unifying components, while it could have been the best approach, the amount of conditions could end up being higher than expected, making it not so useful to be shared components

Reply:
resolved.